src/main/java/org/kohsuke/github/Requester.java

```diff
@@ -327,89 +379,80 @@ public Requester inBody() {
     /**
      * To.
      *
-     * @param tailApiUrl
-     *            the tail api url
      * @throws IOException
      *             the io exception
      */
-    public void to(String tailApiUrl) throws IOException {
-        _to(tailApiUrl, null, null);
+    public void to() throws IOException {
+        _to(null, null);
     }
 
     /**
      * Sends a request to the specified URL, and parses the response into the given type via databinding.
      *
      * @param <T>
      *            the type parameter
-     * @param tailApiUrl
-     *            the tail api url
      * @param type
      *            the type
      * @return {@link Reader} that reads the response.
      * @throws IOException
      *             if the server returns 4xx/5xx responses.
      */
-    public <T> T to(String tailApiUrl, Class<T> type) throws IOException {
-        return _to(tailApiUrl, type, null);
+    public <T> T to(@Nonnull Class<T> type) throws IOException {
+        return _to(type, null);
     }
 
     /**
-     * Like {@link #to(String, Class)} but updates an existing object instead of creating a new instance.
+     * Sends a request to the specified URL, and parses the response into the given type via databinding.
+     *
+     * @param <T>
+     *            the type parameter
+     * @param type
+     *            the type
+     * @return {@link Reader} that reads the response.
+     * @throws IOException
+     *             if the server returns 4xx/5xx responses.
+     */
+    public <T> T[] toArray(@Nonnull Class<T[]> type) throws IOException {
```

Comment:
Overloading `to()` doesn't work with `T[]` and `T`.  The compiler can't handle it.  Instead moved this logic to this new method.  

This results in cleaner and clearer code - there is no case where the caller doesn't already know whether they want an array or an instance returned so why call one method and then immediately have an `if` statement.